modules/launcher/src/test/java/org/mule/module/launcher/DeploymentServiceTestCase.java

```diff
@@ -2865,6 +2867,37 @@ private void assertApplicationDeploymentSuccess(DeploymentListener listener, Str
         assertDeploymentSuccess(listener, artifactName);
         assertStatus(artifactName, ApplicationStatus.STARTED);
     }
+    
+    private void assertApplicationDeploymentSuccessMulltipleArtifacts(DeploymentListener listener, String... artifactNames)
+    {
+        assertDeploytmentSuccessMultipleArtifacts(listener, artifactNames);
+        for (String artifactName : artifactNames)
+        {
+            assertStatus(artifactName, ApplicationStatus.STARTED);
+        }
+    }
+
+    private void assertDeploytmentSuccessMultipleArtifacts(final DeploymentListener listener, final String... artifactNames)
+    {
+        Prober prober = new PollingProber(DEPLOYMENT_TIMEOUT, 100);
+        prober.check(new JUnitProbe()
+        {
+            @Override
+            protected boolean test() throws Exception
+            {
+                ArgumentCaptor<String> arguments = forClass(String.class);
+                verify(listener, times(artifactNames.length)).onDeploymentSuccess(arguments.capture());
```

Comment:
I think this is the problems which renders this test flaky in jenkins. Sometimes the second app is deployed before the first one, so the verify fails because of the argument used.

This is the trace:

java.lang.AssertionError: Failed to deploy application: dummy-domain-app1

Argument(s) are different! Wanted:
deploymentListener.onDeploymentSuccess(
    "dummy-domain-app1"
);
-> at org.mule.module.launcher.DeploymentServiceTestCase$29.test(DeploymentServiceTestCase.java:2877)
Actual invocation has different arguments:
deploymentListener.onDeploymentSuccess(
    "dummy-domain-app2"
);
-> at org.mule.module.launcher.CompositeDeploymentListener.onDeploymentSuccess(CompositeDeploymentListener.java:59)

	at org.mule.tck.probe.PollingProber.check(PollingProber.java:33)
	at org.mule.module.launcher.DeploymentServiceTestCase.assertDeploymentSuccess(DeploymentServiceTestCase.java:2872)
	at org.mule.module.launcher.DeploymentServiceTestCase.assertApplicationDeploymentSuccess(DeploymentServiceTestCase.java:2865)
	at org.mule.module.launcher.DeploymentServiceTestCase.redeploysDomainWithOneApplicationFailedOnFirstDeployment(DeploymentServiceTestCase.java:2487)